packages/glossary-plugin/src/plugin.test.ts

```diff
@@ -17,27 +21,77 @@ describe("LARA plugin initialization", () => {
 });
 
 describe("GlossaryPlugin", () => {
-  it("renders PluginApp component", () => {
+  it("renders PluginApp component", async () => {
     const context = {
       authoredState: "{}",
       learnerState: "",
       pluginId: "123",
       div: document.createElement("div")
     };
     const plugin = new GlossaryPlugin(context);
-    expect(plugin.pluginAppComponent).not.toBeNull();
+    // Note that this function doesn't have to be called manually in most cases. Constructor does it,
+    // but it does not wait for it to complete (as it can't). So, we need to do it by hand while testing.
+    await plugin.renderPluginApp();
+    expect(plugin.pluginAppComponent).not.toBeUndefined();
   });
 
-  it("provides reasonable fallback state if provided values are malformed", () => {
+  it("provides reasonable fallback state if provided values are malformed", async () => {
     const context = {
       authoredState: "some old unsupported format",
       learnerState: "some old unsupported format",
       pluginId: "123",
       div: document.createElement("div")
     };
     const plugin = new GlossaryPlugin(context);
+    // Note that this function doesn't have to be called manually in most cases. Constructor does it,
+    // but it does not wait for it to complete (as it can't). So, we need to do it by hand while testing.
+    await plugin.renderPluginApp();
```

Comment:
Doesn't this result in` plugin.renderPluginApp()` getting called twice in these tests -- once from the constructor, and once explicitly?  Maybe it doesn't matter, but it seems like it might be a point of confusion.  

Your notes here are helpful, as people hopefully won't mistakenly invoke `.renderPluginApp()` directly.  

I am wondering if there is another way to wait for signs that `renderPluginApp` has been invoked here, instead of re-invoking with an `await`.

Reply:
> Doesn't this result in plugin.renderPluginApp() getting called twice in these tests -- once from the constructor, and once explicitly? 

It does. But I've made sure that this function is idempotent, so it doesn't matter. I've updated comments to make it more explicit.

> I am wondering if there is another way to wait for signs that renderPluginApp has been invoked here, instead of re-invoking with an await.

I can't really see it. The constructor can't be an async function and we can't wait for it.